diff --git a/arch/arm64/Makefile b/arch/arm64/Makefile
index 97d8017d0..18f30d1d1 100644
--- a/arch/arm64/Makefile
+++ b/arch/arm64/Makefile
@@ -62,8 +62,8 @@ ifeq ($(cc-name),clang)
 # This is a workaround for https://bugs.llvm.org/show_bug.cgi?id=30792.
 # TODO: revert when this is fixed in LLVM.
 KBUILD_CFLAGS	+= -mno-implicit-float
-else
-KBUILD_CFLAGS	+= -mgeneral-regs-only
+#else
+#KBUILD_CFLAGS	+= -mgeneral-regs-only
 endif
 KBUILD_CFLAGS	+= $(lseinstr) $(brokengasinst)
 KBUILD_CFLAGS	+= -fno-asynchronous-unwind-tables
